rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,453 @@
+  * CN: Common Name
+  * L: Locality
+  * ST: State
+  * O: Organisation
+  * OU: Organisation Unit
+  * STREET: Street Address
+* Subject Other Name (san)
+  * email
+  * dns
+  * uri
+* Extended Key Usage (eku)
+  * Any OID
+* A Free-to-Use CA For Code Signing (fulcio-issuer)
+  * Any issuer hostname
+
+### Extending the x509 specification
+
+This RFC extends the Subject Other Name (san) attribute with the following attribute:
+
+* Subject Other Name (san)
+  * otherName: A free-form attribute that can be used to specify any attribute that is not covered by the other
+    attributes.
+
+The otherName attribute can be used to specify extra attributes in a x509 certificate. This attribute is added to the specification of this RFC to cater for the use case where the san.otherName attribute is used in the x509 certificate and plays a role in the identification of the holder of the certificate.
+
+## The X509Credential Structure
+
+An `X509Credential` must conform to the general structure of a W3C Verifiable Credential and conform to the following
+rules:
+
+- The credential MUST be in JWT format.
+- `type`: MUST include `VerifiableCredential` and `X509Credential`.
+- `issuer`: MUST be a valid `did:x509` identifier.
+- `credentialSubject`: MUST only contain fields explicitly present in the `did:x509` DID policies with the fields mapped by each type as a separate map. An example: 
+```
+{
+  "subject": {
+    "O" : "My Organisation"
+  },
+  "san" : {
+    "email" : "info@example.com,"
+  }
+}
+```
+
+The credential subject can be identified by any DID method (e.g. `did:web`) accepted by the credential verifier.
+
+### Example `X509Credential`
+
+Below is an example of an `X509Credential` issued by a `did:x509` DID. The credential subject is identified by a
+`did:web`.
+The first snippet is the JWT header, and the second snippet is the credential payload.
+
+(TODO: check this example)
+
+```json
+{
+  "alg": "PS256",
+  "typ": "JWT",
+  "x5c": [
+    "<base64 encoded leaf-certificate in the DER format>",
+    "<base64 encoded issuer-intermediate-certificate in the DER format>",
+    "<base64 encoded issuer-certificate in the DER format>"
+  ],
+  "x5t": "<thumbprint>",
+  "kid": "did:x509:0:sha256:<hash>::subject:O:Library%20The%20Bookworm::subject:L:Bookland::san:otherName:123#1"
+}
+```
+
+(TODO: Add right JSON-LD context)
+Payload:
+
+```json
+{
+  "vc": {
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1"
+    ],
+    "type": [
+      "VerifiableCredential",
+      "X509Credential"
+    ],
+    "issuer": "did:x509:0:sha256:<hash>::subject:O:Library%20The%20Bookworm::subject:L:Bookland::san:otherName:123",
+    "issuanceDate": "2024-12-01T00:00:00Z",
+    "credentialSubject": {
+      "id": "did:web:example.com",
+      "subject:O": "Library The Bookworm",
+      "subject:L": "Bookland",
+      "san:otherName": "123"
+    }
+  }
+}
+```
+
+## Validation
+
+To validate an `X509Credential`, the following steps MUST be performed:
```

Comment:
We should add a step that the DID specifies an accepted trust anchor (CA certificate) for `ca-fingerprint`, referring to the Security Consideration